gammapy/irf/edisp/map.py

```diff
@@ -538,3 +540,50 @@ def peek(self, figsize=(15, 5)):
 
         """
         self.get_edisp_kernel().peek(figsize)
+
+
+def _drm_to_edisp(filename, geom, geom_true, geom_psf):
+    """Convert fermi drm into edisp"""
+    with fits.open(filename) as hdu:
+        matrix_hdu = hdu["DRM"]
+        data = matrix_hdu.data
+        header = matrix_hdu.header
+
+        pdf_matrix = np.zeros([len(data), header["DETCHANS"]], dtype=np.float64)
+
+        for i, l in enumerate(data):
+            if l.field("N_GRP"):
+                m_start = 0
+                for k in range(l.field("N_GRP")):
+                    chan_min = l.field("F_CHAN")[k] - 1
+                    chan_max = l.field("F_CHAN")[k] + l.field("N_CHAN")[k] - 1
+
+                    pdf_matrix[i, chan_min:chan_max] = l.field("MATRIX")[
+                        m_start : m_start
+                        + l.field("N_CHAN")[
+                            k
+                        ]  # noqa: E203
+                    ]
+                    m_start += l.field("N_CHAN")[k]
+
+        table = Table.read(matrix_hdu)
+    axis = MapAxis.from_table(table, format="ogip-arf")
+
+    if pdf_matrix.shape[1] != len(geom.axes["energy"].center):
+        raise ValueError(
+            "Energy axes of counts and DRM do not match. Check that edisp_bins are the same"
+        )
+    if pdf_matrix.shape[0] != len(geom_true.axes["energy_true"].center) + 1:
+        raise ValueError(
+            "Energy true axes of exposure and DRM do not match. Check that edisp_bins are the same"
+        )
+
+    energy = geom.axes["energy"]
+    energy_true = axis.copy(name="energy_true", unit="MeV")
```

Comment:
Is the DRM really always a square matrix? I guess not, it depends on the `edisp_bins` parameter. I have checked and the HDU contains the `DETCHANS` keyword, which tells you the number of channels in the "reco" axis. Bases on this information you should be able to construct both axes correctly. As far as I understood, the "true" axis is aligned with the "reco" axis, but just contains the additional ("appended") bins.

However this still has the issue, that the `gtdrm` does not output a matrix that aligns with the exposure. So some kind of resampling is needed for the RMF matrix, to have it on the same true energy axis as the exposure.

Reply:
I have implement a "quick and dirty" version here: https://github.com/adonath/snakemake-workflow-fermi-lat/commit/f9d1f32563b752dea8bfa472acf9ee3aa3c1f110#diff-c8c28ef1325ba68f3569569e41d0f245f8a80cab8b810a30dd46c3bd2fda9779R21

This might be a good starting point. I would be happy to delete it once the drm file format is supported in Gammapy.